pkg/resources/task.go

```diff
@@ -93,16 +101,10 @@ var taskSchema = map[string]*schema.Schema{
 		},
 	},
 	"config": {
-		Type:     schema.TypeString,
-		Optional: true,
-		DiffSuppressFunc: SuppressIfAny(
-			IgnoreChangeToCurrentSnowflakeValueInShow("config"),
-			// TODO(SNOW-1348116 - next pr): Currently config has to be passed with $$ prefix and suffix. The best solution would be to put there only json, so it could be retrieved from file, etc. Move $$ adding to the SDK.
-			func(k, oldValue, newValue string, d *schema.ResourceData) bool {
-				return strings.Trim(oldValue, "$") == strings.Trim(newValue, "$")
-			},
-		),
-		Description: "Specifies a string representation of key value pairs that can be accessed by all tasks in the task graph. Must be in JSON format.",
+		Type:             schema.TypeString,
+		Optional:         true,
+		DiffSuppressFunc: IgnoreChangeToCurrentSnowflakeValueInShow("config"),
+		Description:      "Specifies a string representation of key value pairs that can be accessed by all tasks in the task graph. Must be in JSON format.",
```

Comment:
Can we add it to the config that `$$` is added automatically?

Reply:
I mean, does it matter what happens underneath on such low level? I'm not sure, I would say the most important info is that the accepted format should be in JSON.

Reply:
But someone may think that it needs to be wrapped in `$$`, and then the config would be double-wrapped, right?